### ISSUE TITLE
[vLLM] torch.compile on sample_from_logits produces incorrect output under SPMD/TP

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3589]
## Problem

Compiling `sample_from_logits` with `torch.compile(backend="tt")` under SPMD tensor parallel mode causes all tokens to be token_id 0. The function works correctly when run eagerly under TP, and works correctly when compiled without TP.

The original code had a conditional that skipped compilation under TP with this TODO:
```python
# TODO: Under SPMD mode, sample_from_logits has correctness issue.
#       Re-enable the torch.compile once the issue is fixed in torchxla.
```

An attempt was made to remove this guard (reasoning that logits are already replicated before reaching sampling), but the issue persists even with explicit `sharding_constraint_tensor` replication before the compiled function.

## Reproduction

With `enable_tensor_parallel: True` on n300 (Llama-3.2-3B or Qwen3-0.6B):

1. Add `@torch.compile(backend="tt", fullgraph=True, dynamic=False)` to `sample_from_logits`
2. Run `test_greedy_determinism[n300]`
3. Output is `!!!!!!!!!!!!!!!!!!!!` (all token_id 0)
4. Remove the decorator → output is correct

## Current workaround

`sample_from_logits` is compiled conditionally — only when `enable_tensor_parallel` is False. This means TP mode has an eager fallback in the sampling hot path.

## What was ruled out

- **Logit sharding**: Adding unconditional `sharding_constraint_tensor(logits, mesh, (None, None))` before `sample_from_logits` did not fix the issue
- **Model-specific**: Reproduces with both Llama-3.2-3B and Qwen3-0.6B (both tied-embedding models)
- **`ParallelLMHead` vs `VocabParallelEmbedding`**: Both models use tied embeddings, so the `ParallelLMHead` sharding constraint path is never entered

## Expected fix

Investigate the interaction between `torch.compile(backend="tt")` and XLA SPMD partitioning in the sampling graph (`argmax`, `multinomial`, temperature scaling, top-k/top-p filtering). The compiled graph likely has incorrect sharding propagation for one of these ops.
